common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt

```diff
@@ -356,6 +369,40 @@ class NetworkSensorManager : SensorManager {
         )
     }
 
+    @RequiresApi(Build.VERSION_CODES.M)
+    private fun updateIP6Sensor(context: Context) {
+        if (!isEnabled(context, ip6Addresses)) {
+            return
+        }
+        val ipAddressList: List<String> = ArrayList()
+        var totalAddresses = 0
+
+        if (checkPermission(context, ip6Addresses.id)) {
+            val connectivityManager = context.applicationContext.getSystemService<ConnectivityManager>()
+            val activeNetwork = connectivityManager?.activeNetwork
+            val ipAddresses = connectivityManager?.getLinkProperties(activeNetwork)?.linkAddresses
+            if (ipAddresses.isNullOrEmpty()) {
+                return
```

Comment:
`return` does not seem correct here as we need to make sure to actually update the sensor to send an appropriate state if the sensor is enabled with proper permissions

Reply:
copy-paste error. The only return now is if it is disabled